Demo/Demo/WKAppEnvironment+Extensions.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+import Components
+import UIKit
+
+extension WKAppEnvironment {
+    static func updateWithTraitCollection(_ traitCollection: UITraitCollection) {
```

Comment:
I think this is useful and should maybe even eventually be rolled into `WKAppEnvironment` itself at some point in some form. But let's feel it out for a bit first. For some reason I feel like we should avoid extending the `Components` classes from the client app unless it feels hyper client specific, and theme and colors are kind of a gray in my mind because we define them in the `Components` framework itself. 

What do you think?